File: dart/scrabble-score/lib/scrabble_score.dart
int score(String word) {
  final Map<String, int> grader = {
    "AEIOULNRST" : 1,
    "DG": 2,
    "BCMP": 3,
    "FHVWY" : 4,
    "K": 5,
    "JX": 8,
    "QZ": 10,
  };

  var score = 0;

  word.split('').forEach((char) {
    grader.forEach((key, value) {
      if (key.contains(char.toUpperCase())) {
        score += value;
      };
    });
  });

  return score;
}

File: go/scrabble-score/scrabble_score.go
// Package scrabble provides implementations for scoring a scrabble word
package scrabble

import "strings"

// Score calculates the score of word using some mapping rules from
// exercism.io
// It returns the score of the word as an int
func Score(word string) int {
	sm := map[string]int{
		"AEIOULNRST": 1,
		"DG":         2,
		"BCMP":       3,
		"FHVWY":      4,
		"K":          5,
		"JX":         8,
		"QZ":         10,
	}

	sc := 0

	for _, chr := range word {
		for k, v := range sm {
			if strings.Contains(k, strings.ToUpper(string(chr))) {
				sc += v
			}
		}
	}

	return sc
}
